### PR TITLE
wlroots: update to 0.16, labwc: update to 0.6.0

### PR DESCRIPTION
**Pull request by tranzystorek-io**
#### Testing the changes
- I tested the changes in this PR: **briefly**

### CODE REVIEW

**Comment by tranzystorek-io** [comment]
Ping @Johnnynator needed to bump hwids to add the pkgconfig file

File: srcpkgs/hwids/template [hunk 1]
```diff
@@ -1,18 +1,28 @@
...
+
+	vmkdir usr/share/pkgconfig
+	vinstall hwdata.pc.in 644 usr/share/pkgconfig hwdata.pc
+	vsed -i \
+		-e 's|@prefix@|/usr|g' \
+		-e 's|@datadir@|/usr/share|g' \
+		-e 's|@pkgdatadir@|${datadir}/hwdata|g' \
+		-e "s|@VERSION@|${version}|g" \
+		-e "s|@NAME@|hwdata|g" \
+		${DESTDIR}/usr/share/pkgconfig/hwdata.pc
```
Comment (paper42) [inline srcpkgs/hwids/template:27]: Can we run `make hwdata.pc` instead?

Comment (tranzystorek-io) [reply srcpkgs/hwids/template:27]: I'm getting

```
=> hwids-0.364_1: running do_install ...
./configure
Makefile.inc generated. Run the make again.
Makefile:25: Makefile.inc: No such file or directory
make: *** [Makefile:32: Makefile.inc] Error 1
=> ERROR: hwids-0.364_1: do_install: 'make Makefile.inc' exited with 2
=> ERROR:   in do_install() at srcpkgs/hwids/template:2
```

**Comment by cinerea0** [comment]
I'm confused as to why there's another split happening with wlroots. My recollection of the last split is that it was done because wayfire was taking a long time to update to wlroots-0.15; 0.16 was only released six days ago as of me typing this. Are you anticipating that there will be another long lag for only a couple compositors that depend on it? Alternatively, is there a big issue with the current version of labwc that this fixes?

**Comment by tranzystorek-io** [comment]
I see your point that we could wait a bit for other compositors to catch up;

I just went ahead since @ifreund had no objections when we discussed this topic

**Comment by paper42** [comment]
My proposal was to open this PR and keep adding updates to it until most of the major compositors are on 0.16 and then merge it. Last time the update took 3 months until we realized we don't want to keep waiting for every compositor to move to 0.15 and split 0.14. I would like to not wait 3 months just because some packages can not be updated this time. wayfire eventually moved to 0.15, but cage is still on 0.14. I would expect the same thing to happen here, even if just with river and because of issues on our side and not upstream's.

We unfortunately won't be able to update river because it needs zig 0.9 which needs LLVM 13 or later (we have LLVM 12) and the last version of LLVM that we would want to package doesn't build with our version of gcc. The current plan is to finish the gcc update to 12 and then package LLVM 15, then zig 0.9 and then we can update river. We are working on this, but this can take a while.

TLDR: I don't want to merge this right now, we should keep adding updates to this PR and merge it when enough compositors move.

**Comment by ifreund** [comment]
> The current plan is to finish the gcc update to 12 and then package LLVM 15, then zig 0.9 and then we can update river. We are working on this, but this can take a while.

Zig 0.9 requires exactly llvm 13 actually, but I intend for the next river release to depend on Zig 0.10 which depends on LLVM 15 so this shouldn't be an issue. Just leaving this comment here for clarity, the overall plan of action seems reasonable to me.

**Comment by tranzystorek-io** [comment]
I put `hwids` in wslroots hostmakedepends, since meson doesn't seem to correctly add sysroot to pkgconfig for cross builds :/

**Comment by paper42** [comment]
> > The current plan is to finish the gcc update to 12 and then package LLVM 15, then zig 0.9 and then we can update river. We are working on this, but this can take a while.
> 
> Zig 0.9 requires exactly llvm 13 actually, but I intend for the next river release to depend on Zig 0.10 which depends on LLVM 15 so this shouldn't be an issue. Just leaving this comment here for clarity, the overall plan of action seems reasonable to me.

I am aware of that, but updating LLVM is a non-trivial task, so we will probably go straight to the latest available version and skip 13 and 14 which means an LLVM update has to happen after the gcc update.

File: srcpkgs/wayfire/template [hunk 2]
```diff
@@ -1,15 +1,15 @@
 # Template file for 'wayfire'
 pkgname=wayfire
-version=0.7.3
-revision=2
+version=0.7.4
```
Comment (paper42) [inline srcpkgs/wayfire/template:3]: This update seems to be unrelated and should be moved to a separate PR

Comment (tranzystorek-io) [reply srcpkgs/wayfire/template]: Switched to a revbump instead

File: srcpkgs/sway/template [hunk 3]
```diff
@@ -1,12 +1,12 @@
 # Template file for 'sway'
 pkgname=sway
 version=1.7
```
Comment (paper42) [inline srcpkgs/sway/template:3]: Can we bump to 1.8rc1 which uses wlroots 0.16 for testing?

File: srcpkgs/sway/template [hunk 4]
```diff
@@ -1,12 +1,12 @@
...
-revision=1
+revision=2
 build_style=meson
 configure_args="-Dwerror=false -Db_ndebug=false"
 conf_files="/etc/sway/config"
 hostmakedepends="pkg-config wayland-devel scdoc git"
```
Comment (paper42) [inline srcpkgs/sway/template:8]: why is git required here?

Comment (tranzystorek-io) [reply srcpkgs/sway/template:8]: Doesn't seem like it's required; build config uses it to get hash and branch name

File: srcpkgs/sway/template (hunk 3 above)
Comment (tranzystorek-io) [reply srcpkgs/sway/template]: Done

File: srcpkgs/sway/template [hunk 5]
```diff
@@ -1,6 +1,7 @@
 # Template file for 'sway'
 pkgname=sway
-version=1.7
+_rc="1"
+version=1.8
```
Comment (paper42) [inline srcpkgs/sway/template:4]: (I can not comment in the right place, so I will leave the comment here)

* replace pcre-devel with pcre2-devel

File: srcpkgs/sway/template [hunk 6]
```diff
@@ -13,8 +14,8 @@ short_desc="Tiling Wayland compositor compatible with i3"
 maintainer="Olaf Mersmann <olafm@p-value.net>"
 license="MIT"
 homepage="https://swaywm.org"
-distfiles="https://github.com/swaywm/${pkgname}/archive/${version}.tar.gz"
-checksum=0c64536fb2883ec518e75cfb119f9300115fb49b084e3fde62b794fe2c6c8d84
+distfiles="https://github.com/swaywm/${pkgname}/archive/${version}-rc${_rc}.tar.gz"
```
Comment (paper42) [inline srcpkgs/sway/template:17]:
```suggestion
distfiles="https://github.com/swaywm/sway/archive/${version}-rc${_rc}.tar.gz"
```

File: srcpkgs/sway/template (hunk 5 above)
Comment (paper42) [inline srcpkgs/sway/template:4]: according to the [changelog](https://github.com/swaywm/sway/issues/7019#issue-1244738993):

> On Linux, CAP_SYS_NICE can be set to improve scheduling

Is this something we want or do we want to leave this to the user?

File: srcpkgs/wlroots0.15/template [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+ wayland-protocols vulkan-loader Vulkan-Headers xcb-util-errors-devel
+ xcb-util-wm-devel xcb-util-renderutil-devel "
+makedepends="${_devel_depends}
+ libxcb-devel xcb-util-cursor-devel xcb-util-devel xcb-util-image-devel
+ xcb-util-keysyms-devel xcb-util-xrm-devel xorg-server-xwayland"
+short_desc="Modular Wayland compositor library"
+maintainer="Isaac Freund <mail@isaacfreund.com>"
+license="MIT"
+homepage="https://gitlab.freedesktop.org/wlroots/wlroots"
+distfiles="${homepage}/-/releases/${version}/downloads/wlroots-${version}.tar.gz"
```
Comment (paper42) [inline srcpkgs/wlroots0.15/template:22]:
```suggestion
distfiles="https://gitlab.freedesktop.org/wlroots/wlroots/-/releases/${version}/downloads/wlroots-${version}.tar.gz"
```

File: srcpkgs/sway/template (hunk 5 above)
Comment (tranzystorek-io) [reply srcpkgs/sway/template:4]: Done

File: srcpkgs/sway/template (hunk 6 above)
Comment (tranzystorek-io) [reply srcpkgs/sway/template]: Done

File: srcpkgs/wlroots0.15/template (hunk 7 above)
Comment (tranzystorek-io) [reply srcpkgs/wlroots0.15/template]: Done

**Comment by skmpz** [comment]
@tranzystorek-io @paper42 i've managed to build using https://github.com/swaywm/sway/commit/dca0bb5749bc16f91ab964fc1b06ebb9a453368f and tested both with and without CAP_SYS_NICE.. everything looks good so far..

File: srcpkgs/sway/template (hunk 5 above)
Comment (ardadem) [reply srcpkgs/sway/template:4]: I think something we want, kwin has it: https://github.com/void-linux/void-packages/blob/ded277952d79804a1a4ed473e5848f65fe121bda/srcpkgs/kwin/INSTALL

**Comment by tranzystorek-io** [comment]
Bumped sway to 1.8-rc2, should be good to test out

File: srcpkgs/wayfire/template [hunk 8]
```diff
@@ -1,15 +1,15 @@
 # Template file for 'wayfire'
 pkgname=wayfire
 version=0.7.3
```
Comment (paper42) [inline srcpkgs/wayfire/template:3]: 0.7.5 is out and uses wlroots 0.16

File: srcpkgs/sway/template (hunk 5 above)
Comment (tranzystorek-io) [reply srcpkgs/sway/template:3]: added

File: srcpkgs/sway/template (hunk 4 above)
Comment (paper42) [reply srcpkgs/sway/template:8]: in that case, please drop git from hostmakedepends, this is built from a tarball, not a git repository, so it won't work anyway

File: srcpkgs/sway/template (hunk 5 above)
Comment (paper42) [reply srcpkgs/sway/template:3]: also add libcap-progs to $depends

Comment (tranzystorek-io) [reply srcpkgs/sway/template:3]: done

File: srcpkgs/sway/template (hunk 4 above)
Comment (tranzystorek-io) [reply srcpkgs/sway/template]: done

File: srcpkgs/sway/INSTALL [hunk 9]
```diff
@@ -0,0 +1,5 @@
+case "${ACTION}" in
+post)
+	setcap CAP_SYS_NICE=+ep usr/bin/sway
```
Comment (mhmdanas) [inline srcpkgs/sway/INSTALL:3]: Doing this currently breaks loading custom keymaps from a user directory, so it should probably be removed and added back when https://github.com/swaywm/sway/pull/7326 is included in a future release (also see https://github.com/xkbcommon/libxkbcommon/pull/312).

Comment (tranzystorek-io) [reply srcpkgs/sway/INSTALL:3]: Since this has been merged in, could you post an issue for this?

Comment (mhmdanas) [reply srcpkgs/sway/INSTALL:3]: I'll just open a PR when I get the time. :)

Comment (mhmdanas) [reply srcpkgs/sway/INSTALL:3]: #41358
